lib/trino-collect/src/test/java/io/trino/collect/cache/TestEvictableCache.java

```diff
@@ -542,4 +549,55 @@ public void testInvalidateAndLoadConcurrently(Invalidation invalidation)
             executor.awaitTermination(10, SECONDS);
         }
     }
+
+    @Test(dataProvider = "disabledCacheImplementations")
+    public void testPutOnEmptyCache(DisabledCacheImplementation disabledCacheImplementation)
+    {
+        Map<Object, Object> cache = EvictableCacheBuilder.newBuilder()
+                .maximumSize(0)
+                .disabledCacheImplementation(disabledCacheImplementation)
+                .build()
+                .asMap();
+
+        int key = 0;
+        int value = 1;
+        assertThat(cache.put(key, value)).isNull();
+        assertThat(cache.put(key, value)).isNull();
+        assertThat(cache.putIfAbsent(key, value)).isNull();
+        assertThat(cache.putIfAbsent(key, value)).isNull();
+    }
+
+    @Test
+    public void testPutOnNonEmptyCache()
+    {
+        Map<Object, Object> cache = EvictableCacheBuilder.newBuilder()
+                .maximumSize(10)
+                .build()
+                .asMap();
+
+        int key = 0;
+        int value = 1;
+        assertThatThrownBy(() -> cache.put(key, value))
+                .isInstanceOf(UnsupportedOperationException.class)
+                .hasMessage("The operation is not supported, as in inherently races with cache invalidation. Use get(key, callable) instead.");
+        assertThatThrownBy(() -> cache.putIfAbsent(key, value))
+                .isInstanceOf(UnsupportedOperationException.class)
+                .hasMessage("The operation is not supported, as in inherently races with cache invalidation");
+    }
+
+    @DataProvider
+    public static Object[][] disabledCacheImplementations()
+    {
+        return Stream.of(DisabledCacheImplementation.values())
+                .collect(toDataProvider());
+    }
+
+    private static <T> Collector<T, ?, Object[][]> toDataProvider()
```

Comment:
DataProviders#toDataProvider ?

Reply:
It is a copy from it. The one you mentioned is in trino-main. Maybe I can remove the other one?

Reply:
didn't know that we have `lib` folder

Reply:
you mean extracting helpers from trino-main to lib ?

Reply:
I moved this class to `trino-testing-services`. That lib should be loaded in any tests anyway as it provides required services.